client/src/features/action/needs/NeedFormGoods.tsx

```diff
@@ -10,11 +11,22 @@ import {
 } from '../../../assets/sharedComponents/Forms';
 import NeedOfferForm from '../NeedOfferForm';
 
-const categories = [
-  { value: 'figs', text: 'Figs' },
-  { value: 'peaches', text: 'Peaches' },
-  { value: 'pears', text: 'Pears' },
-];
+import type { Option } from '../../../types';
+
+const fetchCategories = async (): Promise<Option[]> => {
+  const res = await fetch('http://localhost:3001/api/categories?applies_to_assets=true');
+  const data = await res.json();
+
+  const categories = await data.map((category: any) => {
+    const value = category.name.toLowerCase();
+    const text = category.name;
+
+    return { value, text };
```

Comment:
i _think_ if we add a `id?: string` to `Option` type and put the id in this object, it'll get passed to BE on form submit and then automatically set the asset<>category relationship